### PR TITLE
Refactor/sealer accepts just what it needs

### PR DESCRIPTION
**Pull request by laser**
Makes incremental progress towards #17 

## Why does this PR exist?

We have to jump through a bunch of hoops to share access to the key-value store and the sector store across threads. That's a bummer, because we don't actually need to perform any mutations in the worker threads.

## What's in this PR?

- drop all of the `Arc`'ed stuff and instead make `Scheduler` thread own all the mutable state
- drop `WrappedKeyValueStore`

### CODE REVIEW

File: sector-builder/src/builder.rs [hunk 1]
```diff
@@ -26,39 +32,23 @@ pub struct SectorBuilder {
...
+    pub fn init_from_metadata(
         sector_class: SectorClass,
         last_committed_sector_id: SectorId,
-        metadata_dir: S,
+        metadata_dir: String,
         prover_id: [u8; 31],
-        sealed_sector_dir: S,
-        staged_sector_dir: S,
+        sealed_sector_dir: String,
+        staged_sector_dir: String,
```
Comment (dignifiedquire) [inline sector-builder/src/builder.rs]: Nit: directories should ideally be `impl AsRef<Path>`

Comment (dignifiedquire) [reply sector-builder/src/builder.rs]: and stored as `PathBuf`

File: sector-builder/src/builder.rs [hunk 2]
```diff
@@ -68,33 +58,69 @@ impl SectorBuilder {
...
+                    .map(Into::into);
+
+            loaded.unwrap_or_else(|| SectorBuilderState {
+                staged: StagedState {
+                    sector_id_nonce: u64::from(last_committed_sector_id),
+                    sectors: Default::default(),
+                },
+                sealed: Default::default(),
+            })
+        };
```
Comment (dignifiedquire) [inline sector-builder/src/builder.rs]: would be nice to wrap this into a method, like `SectorBuilderState::new(last_committed_sector_id)`

File: sector-builder/src/scheduler.rs [hunk 3]
```diff
@@ -162,15 +120,15 @@ impl Scheduler {
...
-    sector_size: PaddedBytesAmount,
+    pub kv_store: T,
+    pub sector_store: S,
+    pub state: SectorBuilderState,
+    pub sealer_input_tx: mpsc::Sender<SealerInput>,
+    pub scheduler_input_tx: mpsc::SyncSender<Request>,
+    pub max_num_staged_sectors: u8,
+    pub max_user_bytes_per_staged_sector: UnpaddedBytesAmount,
+    pub prover_id: [u8; 31],
+    pub sector_size: PaddedBytesAmount,
```
Comment (dignifiedquire) [inline sector-builder/src/scheduler.rs]: maybe `pub(crate)` instead?

File: sector-builder/src/scheduler.rs [hunk 4]
```diff
@@ -300,7 +258,7 @@ impl<T: KeyValueStore, S: SectorStore> SectorMetadataManager<T, S> {
         piece_path: String,
         store_until: SecondsSinceEpoch,
     ) -> Result<SectorId> {
-        let destination_sector_id = add_piece(
+        let destination_sector_id = crate::helpers::add_piece(
```
Comment (dignifiedquire) [inline sector-builder/src/scheduler.rs]: "good rust rule": import the top level, e.g. in this case

```rust
use crate::helpers;

helpers::add_piece()
```

File: sector-builder/src/scheduler.rs [hunk 5]
```diff
@@ -386,25 +344,77 @@ impl<T: KeyValueStore, S: SectorStore> SectorMetadataManager<T, S> {
...
-                Err(err) => {
-                    if let Some(staged_sector) = staged_state.sectors.get_mut(&sector_id) {
-                        staged_sector.seal_status =
-                            SealStatus::Failed(format!("{}", err_unrecov(err)));
+            let staged_sector = staged_state
+                .sectors
+                .get_mut(&sector_id)
+                .expect("missing staged sector");
+
+            let _ = result
```
Comment (dignifiedquire) [inline sector-builder/src/scheduler.rs]: why are you throwing away the result?

File: sector-builder/src/scheduler.rs (hunk 3 above)
Comment (laser) [reply sector-builder/src/scheduler.rs]: good call

File: sector-builder/src/scheduler.rs (hunk 5 above)
Comment (laser) [reply sector-builder/src/scheduler.rs]: I can't do anything with the `result` at this point. I've already run all side effects via lambdas passed to `map_err` and `map`.

File: sector-builder/src/builder.rs (hunk 1 above)
Comment (laser) [reply sector-builder/src/builder.rs]: good call. updated.

File: sector-builder/src/builder.rs (hunk 2 above)
Comment (laser) [reply sector-builder/src/builder.rs]: good call. updated.

File: sector-builder/src/scheduler.rs (hunk 4 above)
Comment (laser) [reply sector-builder/src/scheduler.rs]: done!

File: sector-builder/src/scheduler.rs (hunk 5 above)
Comment (laser) [reply sector-builder/src/scheduler.rs]: I refactored such that I didn't have to throw away the result :)

**Pull request merged**